pkg/virt-controller/services/template.go

```diff
@@ -1547,44 +1546,64 @@ func getIfaceByName(vmi *v1.VirtualMachineInstance, name string) *v1.Interface {
 	return nil
 }
 
-func getCniAnnotations(vmi *v1.VirtualMachineInstance) (cniAnnotations map[string]string, err error) {
-	ifaceListMap := make([]map[string]string, 0)
-	cniAnnotations = make(map[string]string, 0)
+func getMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
+	var ifaceListMap []map[string]string
+
+	relevantNetworks := filterMultusNonDefaultNetworks(vmi)
+	for i, network := range relevantNetworks {
+		iface := getIfaceByName(vmi, network.Name)
+		namespace, networkName := getNamespaceAndNetworkName(vmi, network.Multus.NetworkName)
+		serializedInterface := mapifyMultusInterface(iface, networkName, namespace, i)
+		ifaceListMap = append(ifaceListMap, serializedInterface)
+	}
+
+	if len(ifaceListMap) > 0 {
+		ifaceJsonString, err := serializeMultusInterfaces(ifaceListMap)
+		if err != nil {
+			return "", err
+		}
+		return ifaceJsonString, nil
+	}
+	return "", nil
+}
+
+func serializeMultusInterfaces(ifaceListMap []map[string]string) (string, error) {
```

Comment:
I do not understand what this function adds here.
It saves no lines per what I see (the caller still needs to handle the error).

Reply:
OK.